### PR TITLE
[Merged by Bors] - chore(Polynomial/Derivative): golf derivative_mul

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Polynomial/Derivative.lean [hunk 1]
```diff
@@ -290,40 +290,21 @@ set_option linter.uppercaseLean3 false in
...
+theorem derivative_mul {f g : R[X]} : derivative (f * g) = derivative f * g + f * derivative g := by
+  induction f using Polynomial.induction_on'
+  · simp only [add_mul, map_add, add_assoc, add_left_comm, *]
+  induction g using Polynomial.induction_on'
+  · simp only [mul_add, map_add, add_assoc, add_left_comm, *]
+  rename_i m a n b
+  simp only [monomial_mul_monomial, derivative_monomial]
+  simp only [mul_assoc, (Nat.cast_commute _ _).eq, Nat.cast_add, mul_add, map_add]
+  cases m
+  · simp only [zero_add, Nat.cast_zero, mul_zero, map_zero]
```
Comment (jcommelin) [inline Mathlib/Data/Polynomial/Derivative.lean:302]:
```suggestion
  · simp only [zero_add, Nat.cast_zero, mul_zero, map_zero, Nat.zero_eq]
```

File: Mathlib/Data/Polynomial/Derivative.lean [hunk 2]
```diff
@@ -290,40 +290,21 @@ set_option linter.uppercaseLean3 false in
...
+  cases n
+  · simp only [add_zero, Nat.cast_zero, mul_zero, map_zero]
```
Comment (jcommelin) [inline Mathlib/Data/Polynomial/Derivative.lean:304]:
```suggestion
  · simp only [add_zero, Nat.cast_zero, mul_zero, map_zero, Nat.zero_eq]
```

File: Mathlib/Data/Polynomial/Derivative.lean [hunk 3]
```diff
@@ -290,40 +290,21 @@ set_option linter.uppercaseLean3 false in
...
+  rename_i m n
+  simp only [Nat.add_succ_sub_one, add_tsub_cancel_right]
```
Comment (jcommelin) [inline Mathlib/Data/Polynomial/Derivative.lean:306]:
```suggestion
  simp only [Nat.add_succ_sub_one, add_tsub_cancel_right, Nat.zero_eq]
```

File: Mathlib/Data/Polynomial/Derivative.lean [hunk 4]
```diff
@@ -290,40 +290,21 @@ set_option linter.uppercaseLean3 false in
...
-        congr
-        · rw [← sum_C_mul_X_pow_eq g]
-        · rw [← sum_C_mul_X_pow_eq f]
-      simp only [sum, sum_add_distrib, Finset.mul_sum, Finset.sum_mul, derivative_apply]
-      simp_rw [← smul_mul_assoc, smul_C, nsmul_eq_mul']
-      rw [Finset.sum_comm]
-      congr 1
-      rw [Finset.sum_comm]
+theorem derivative_mul {f g : R[X]} : derivative (f * g) = derivative f * g + f * derivative g := by
+  induction f using Polynomial.induction_on'
```
Comment (ocfnash) [inline Mathlib/Data/Polynomial/Derivative.lean:294]: I slightly prefer using `induction' ... with ...` to avoid the `rename_i`s below.

**Review by ocfnash — APPROVED** [review]
Thanks!

One optional suggestion.

bors d+

**Review by jcommelin — COMMENTED** [review]
bors merge

**Pull request closed without merging**